cmd/internal/env/env_test.go

```diff
@@ -0,0 +1,177 @@
+	}
+	root.Run(root, []string{})
+	out := rootWriter.String()
+	expectation := "3; ; false"
+	if out != expectation {
+		t.Fatalf("expected flag values %q, got %q", expectation, out)
+	}
+}
+
+func TestCmdFlagsImpl_CheckEnvironmentVariables_AllEnvVarsSingleCommand(t *testing.T) {
+	rootWriter := bytes.NewBuffer([]byte{})
+	root := mockRootCmd(rootWriter)
+	t.Setenv("OPA_INT", "40")
+	t.Setenv("OPA_SOME_STRING", "test")
+	t.Setenv("OPA_BOOL", "true")
+	if err := root.PreRunE(root, []string{}); err != nil {
+		t.Fatalf("unexpected error: %s", err.Error())
+	}
+	root.Run(root, []string{})
+	out := rootWriter.String()
+	expectation := "40; test; true"
+	if out != expectation {
+		t.Fatalf("expected flag values %q, got %q", expectation, out)
+	}
+}
+
+func TestCmdFlagsImpl_CheckEnvironmentVariables_ChildCommandAllEnvVars(t *testing.T) {
+	root := mockRootCmd(&bytes.Buffer{})
+	childWriter := bytes.NewBuffer([]byte{})
+	child := mockChildCmd(childWriter)
+	root.AddCommand(child)
+	t.Setenv("OPA_CHILD_SECOND_INT", "7")
+	t.Setenv("OPA_CHILD_SECOND_STRING", "testing child")
+	t.Setenv("OPA_CHILD_SECOND_BOOL", "false")
+	if err := child.PreRunE(child, []string{}); err != nil {
+		fmt.Fprintf(os.Stdout, "unexpected error: %s", err.Error())
+		t.Fatalf("BONK")
+	}
+	child.Run(child, []string{})
+	childOut := childWriter.String()
+	childExpectation := "7; testing child; false"
+	if childOut != childExpectation {
+		t.Fatalf("expected child flag values %q, got %q", childExpectation, childOut)
+	}
+}
+
+func TestCmdFlagsImpl_CheckEnvironmentVariables_ChildCommandReturnsSingleErr(t *testing.T) {
+	root := mockRootCmd(&bytes.Buffer{})
+	child := mockChildCmd(&bytes.Buffer{})
+	root.AddCommand(child)
+	t.Setenv("OPA_CHILD_SECOND_BOOL", "7")
+	err := child.PreRunE(child, []string{})
+	if err == nil {
+		t.Fatalf("expected error, found none")
+	}
+	expectedString := "invalid argument \"7\""
+	if !strings.Contains(err.Error(), expectedString) {
+		t.Fatalf("expected error to include %q, instead got %q", expectedString, err.Error())
+	}
+}
+
+func TestCmdFlagsImpl_CheckEnvironmentVariables_ChildCommandReturnsMultipleErr(t *testing.T) {
+	root := mockRootCmd(&bytes.Buffer{})
+	child := mockChildCmd(&bytes.Buffer{})
+	root.AddCommand(child)
+	t.Setenv("OPA_CHILD_SECOND_INT", "true")
+	t.Setenv("OPA_CHILD_SECOND_BOOL", "7")
+	err := child.PreRunE(child, []string{"child"})
+	expectedString := "invalid argument"
+	if err == nil {
+		t.Fatalf("expected error, found none")
+	}
+	if !strings.Contains(err.Error(), expectedString) {
+		t.Fatalf("expected error to include %q, instead got %q", expectedString, err.Error())
+	}
+	if !strings.Contains(err.Error(), "7") {
+		t.Fatalf("expected error for invalid int 7 as argument for boolean flag")
+	}
+	if !strings.Contains(err.Error(), "true") {
+		t.Fatalf("expected error for invalid int 7 as argument for int flag")
+	}
+}
+
+func TestCmdFlagsImpl_CheckEnvironmentVariables_ConfirmCommandFlagPrecedence(t *testing.T) {
```

Comment:
Good!

Reply:
100% meant to remove that and kept thinking I would come back to it.  Just pushed a new commit, thanks for catching that.